### PR TITLE
Bump ReCirq to Cirq 0.15

### LINKED ISSUE #310: Bump CI to Cirq 0.15
Cirq 0.15 has been released and things have been fixed, xref #308 #306 #309 

The CI needs to learn that current=0.15. There's also a couple of shims in #308 for cirq 0.13 that we can remove

cc @verult @dstrain115

### PR DESCRIPTION
**Pull request by mpharrigan**
Fixes #310 

I've also fixed a couple of rote deprecation issues but not all of them for 1.0

### CODE REVIEW

File: recirq/qaoa/gates_and_compilation.py [hunk 1]
```diff
@@ -557,28 +558,26 @@ def compile_single_qubit_gates(
         mutate: By default, return a copy of the circuit. Otherwise,
             mutate in place.
     """
-    if mutate:
-        c2 = circuit
-    else:
-        c2 = circuit.copy()
+    assert not mutate, "cirq doesn't mutate."
```
Comment (dstrain115) [inline recirq/qaoa/gates_and_compilation.py:561]: Can we piggyback on Cirq's deprecation functionality and deprecate this paramete?
Same below.

**Review by dstrain115 — APPROVED** [review]
Looks good so far minus test failures and one comment.

File: recirq/qaoa/gates_and_compilation.py (hunk 1 above)
Comment (mpharrigan) [reply recirq/qaoa/gates_and_compilation.py]: removed the vestigial arguments entirely

**Comment by mpharrigan** [comment]
On the basis that this gets us to 0.15 and is holding up work to get us to 1.0, I'm going to merge once the cirq current (i.e. 0.15) and previous tests pass and change the cirq-next (i.e. 1.0) test to be optional for now.

**Comment by dstrain115** [comment]
SGTM.

**Pull request merged**